### ISSUE TITLE
Potentially unsafe array access after the array length asserted

### ISSUE THREAD

**Issue by Bohdankm22** [typetools/checker-framework#1158]
Consider following code:

```
private static void readInfo(String k, int m) {
        String[] parts = k.split(" ");
        assert parts.length > 2;
        // The array length has to be more than 2
        Integer.parseInt(parts[0]);    // Potentially unsafe array access: the index could be larger than the arrays bound
        Integer.parseInt(parts[1]);    // Potentially unsafe array access: the index could be larger than the arrays bound
        Integer.parseInt(parts[2]);    // Potentially unsafe array access: the index could be larger than the arrays bound
    }
```

Each of the array accesses leads to the following error:
```
Error:(120, 32) java: [array.access.unsafe.high] Potentially unsafe array access: the index could be larger than the array's bound
  found   : @UpperBoundUnknown int
  required: an integer less than parts's length (@LTLengthOf("parts"))
```

Expected result: as the array length was asserted to be more than some value is safe to access the array with indexes less than the array length.

**Comment by wmdietl** [comment]
Did you try https://checkerframework.org/manual/#assumeassertion
We don't use all assertions for type refinement.

**Comment by mernst** [comment]
This code passes the type-checker without warnings:

```java
public class Assertion {

  private static void readInfo(String k, int m) {
    String[] parts = k.split(" ");
    assert parts.length > 2 : "@AssumeAssertion(index)";
    // The array length has to be more than 2
    Integer.parseInt(parts[0]);
    Integer.parseInt(parts[1]);
    Integer.parseInt(parts[2]);
  }

}
```

Therefore, it looks like the checker is working correctly.

Thanks for reporting this, though, and let us know if there is something we could add to the manual to make this behavior more clear.  (Searching for "assert" in the manual should turn up the documentation Werner pointed you at.)
